examples/plugins/job-pre-post/mail/avocado_job_mail.py

```diff
@@ -30,23 +96,145 @@ def __init__(self):
+                out.write("          %s\n" % t['fail_reason'])
+            idx = idx + 1
+
+        out.write("RESULTS    : PASS %d | ERROR %d | FAIL %d | SKIP %d\n" %
+                  (j['pass'], j['errors'], j['failures'], j['skip']))
+        out.write('TESTS TIME : %.2f s\n' % j['time'])
+
+        return MIMEText(out.getvalue())
+
+    def _try_attach(self, mime, path, mime_type='octet-stream'):
+        if os.path.exists(path):
+            self.log.debug('try attach %s' % path)
+            fp = open(path, 'rb')
+            if not mime_type == 'text':
+                part = MIMEBase('application', mime_type)
+                part.set_payload(fp.read())
+                if mime_type == 'octet-stream':
+                    encoders.encode_base64(part)
+                desc = 'attachment; filename="%s"' % os.path.basename(path)
+                part.add_header('Content-Disposition', desc)
+            else:
+                part = MIMEText(fp.read())
+            fp.close()
+            mime.attach(part)
+        else:
+            self.log.debug('try_attach %s not exits' % path)
+
+    def mail_report(self, job):
+
+        self.parse_job_opts(job)
+        if not self.enabled:
+            return
+
+        msg = MIMEMultipart('text')
         msg['From'] = self.sender
         msg['To'] = self.rcpt
+        msg['Cc'] = ', '.join(self.rcpt_cc)
+
+        if self.body_type == 'minimal':
+            # build proper subject based on job status
+            self.subject = '%s Job %s - Status: %s' % (self.subject,
+                                                       job.unique_id,
+                                                       job.status)
+        else:
+            #part1 = MIMEMessage('text', 'plain')
+            #part1 = MIMEMessage('text')
+            json_path = os.path.join(job.logdir, 'results.json')
+
+            try:
+                fp = open(json_path, 'rb')
+                self.results = json.loads(fp.read())
+                self.subject = '%s Job %s - %s : ' % (self.subject,
+                                                      job.unique_id[:7],
+                                                      job.status)
+                fp.close()
+                msg['Subject'] = self.subject + self._make_stats()
+                body = self._make_body()
+                msg.attach(body)
+                # Jobscripts plugin or tests may generate extra data for us
+                email_report = os.path.join(job.logdir, 'email_report')
+                self._try_attach(msg, os.path.join(job.logdir, 'email_report'), 'text')
+                self._try_attach(msg, json_path, 'json')
+            except Exception as e:
+                self.log.error('Can not parse results.json')
+                self.log.error(e)
+                return
+
+        if self.body_type == 'full':
+            self._try_attach(msg, os.path.join(job.logdir, 'results.html'))
+            self._try_attach(msg, job.logdir + '.zip')
```

Comment:
So this requires the usage of `--archive`, right? Maybe the init should enable it if not already enabled (in case `full` body type is enabled)